src/shared/featureToggle.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SettingsConfiguration } from './settingsConfiguration'
+
+// This enum represents the list of active feature flags to check for
+// Features all need to be added with matching enums and string values
+// e.g. `Feature1 = 'Feature1'`
+// Features that do not follow this scheme will not work.
+// You cannot have more active features than FeatureToggle.maxFeatures (default: 5)
+// Any features that are flagged in the code but not added here will always return false.
+export enum ActiveFeatureKeys {
+}
+
+/**
+ * This class handles feature access for unreleased or gated features.
+ * Example: adding a feature called `NewFeature1`:
+ *
+ * Add the following to the `featureToggle.ActiveFeatureKeys` enum:
+ *
+ * ```
+ * export enum ActiveFeatureKeys {
+ *     NewFeature1 = 'NewFeature1'
+ * }
+ * ```
+ *
+ * You can then gate your feature with the following code (returning a boolean):
+ *
+ * ```
+ * FeatureToggle.isFeatureActive(featureToggle.ActiveFeatureKeys.NewFeature1)
+ * ```
+ *
+ * Finally, you can access the gated feature by adding the following snippet to the VS Code `settings.json` file:
+ *
+ * ```
+ *     "aws.experimentalFeatureFlags": [
+ *         "NewFeature1",
+ *     ],
+ * ```
+ */
+export class FeatureToggle {
```

Comment:
this should be pluralized right? This encompasses the state of all feature toggles.

Reply:
Could go either way. Grammatically, I think it would make more sense as something like `FeatureToggles.Feature1`, but since we're using the enum and `isFeatureActive()`, it's less of a container of toggles.

Reply:
Wait - I thought that was the `ActiveFeature` enum...

Reply:
Just to make sure we're on the same page, the end call will be `FeatureToggle.isFeatureActive(ActiveFeatureKeys.NewFeature1)`. Current proposal is to not have `FeatureToggle` contain a property/function for each individual feature since I don't want to have people edit the `FeatureToggle` class directly.